src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -689,6 +689,207 @@ protected override void ProcessRecord()
+
+                status = NativeMethods.QueryServiceConfig2W(
+                    hService,
+                    NativeMethods.SERVICE_CONFIG_DESCRIPTION,
+                    lpBuffer,
+                    bufferSizeNeeded,
+                    out bufferSizeNeeded);
+
+                if(!status) {
+                    lastError = Marshal.GetLastWin32Error();
+                    Win32Exception exception = new Win32Exception(lastError);
+                    WriteNonTerminatingError(
+                        service,
+                        exception,
+                        "CouldNotGetServiceInfo",
+                        ServiceResources.CouldNotGetServiceInfo,
+                        ErrorCategory.PermissionDenied);
+                }
+                NativeMethods.SERVICE_DESCRIPTIONW description = (NativeMethods.SERVICE_DESCRIPTIONW)Marshal.PtrToStructure(lpBuffer, typeof(NativeMethods.SERVICE_DESCRIPTIONW));
+
+                // Delayed auto start information
+                lpBuffer = IntPtr.Zero;
+                bufferSizeNeeded = 0;
+                status = NativeMethods.QueryServiceConfig2W(
+                    hService: hService,
+                    dwInfoLevel: NativeMethods.SERVICE_CONFIG_DELAYED_AUTO_START_INFO,
+                    lpBuffer: lpBuffer,
+                    cbBufSize: 0,
+                    pcbBytesNeeded: out bufferSizeNeeded);
+
+                lastError = Marshal.GetLastWin32Error();
+                if(lastError != NativeMethods.ERROR_INSUFFICIENT_BUFFER) {
+                    Win32Exception exception = new Win32Exception(lastError);
+                    WriteNonTerminatingError(
+                        service,
+                        exception,
+                        "CouldNotGetServiceInfo",
+                        ServiceResources.CouldNotGetServiceInfo,
+                        ErrorCategory.PermissionDenied);
+
+                }
+                lpBuffer = Marshal.AllocHGlobal((int)bufferSizeNeeded);
+
+                status = NativeMethods.QueryServiceConfig2W(
+                    hService,
+                    NativeMethods.SERVICE_CONFIG_DELAYED_AUTO_START_INFO,
+                    lpBuffer,
+                    bufferSizeNeeded,
+                    out bufferSizeNeeded);
+                if(!status) {
+                    lastError = Marshal.GetLastWin32Error();
+                    Win32Exception exception = new Win32Exception(lastError);
+                    WriteNonTerminatingError(
+                        service,
+                        exception,
+                        "CouldNotGetServiceInfo",
+                        ServiceResources.CouldNotGetServiceInfo,
+                        ErrorCategory.PermissionDenied);
+                }
+                NativeMethods.SERVICE_DELAYED_AUTO_START_INFO autostartInfo = (NativeMethods.SERVICE_DELAYED_AUTO_START_INFO)Marshal.PtrToStructure(lpBuffer, typeof(NativeMethods.SERVICE_DELAYED_AUTO_START_INFO));
```

Comment:
I don't have a strong opinion on this one, but my preference is to have it as part of StartupType similar to what you see in services.msc

Reply:
Can we extend Enums using PowerShell somehow?
C# does not seem to be able to do it using just inheritance.
Currently the [ServiceStartMode][0] class does not support `Automatic (Delayed start)`. 
I could also make my own class, but that would be a breaking change, correct?

[0]: https://msdn.microsoft.com/en-us/library/system.serviceprocess.servicestartmode(v=vs.110).aspx

Reply:
Seems like having it separate would be simplest.